src/ralph/backends/data/base.py

```diff
@@ -317,10 +318,14 @@ def read(  # noqa: PLR0913
             yield from statements
             return
 
+        if not max_statements:
+            return
+
+        max_statements -= 1
         for i, statement in enumerate(statements):
+            yield statement
             if i >= max_statements:
                 return
```

Comment:
Good point) The purpose of the first `if not max_statements:` was to avoid iterating over the statements when `max_statements==0`. However, this is of little use( maybe we should change that to the same behavior of `max_statements=None` - retrieving all statements? WDYT?